hw/opentitan/ot_spi_device.c

```diff
@@ -1527,6 +1485,350 @@ static void ot_spi_device_flash_resume_read(void *opaque)
     qemu_chr_fe_accept_input(&s->chr);
 }
 
+static bool
+ot_spi_device_flash_command_is_filter(OtSPIDeviceState *s, uint8_t cmd)
+{
+    return (bool)(s->spi_regs[R_CMD_FILTER_0 + (cmd / REG_BITS)] &
+                  (1u << (cmd % REG_BITS)));
+}
+
+static uint8_t ot_spi_device_swap_byte_data(
+    uint8_t byte, unsigned byte_sel, uint32_t swap_mask, uint32_t swap_data)
+{
+    g_assert(byte_sel < 4u);
+    uint8_t mask = (uint8_t)((swap_mask >> (byte_sel * 8u)) & 0xffu);
+    uint8_t data = (uint8_t)((swap_data >> (byte_sel * 8u)) & 0xffu);
```

Comment:
Nit: `(uint8_t)(x & 0xffu)` is redundant:

```suggestion
    uint8_t mask = (uint8_t)(swap_mask >> (byte_sel * 8u));
    uint8_t data = (uint8_t)(swap_data >> (byte_sel * 8u));
```